Incompatibility with uvicorn>=0.3.26: "ImportError: cannot import name 'get_logger' from 'uvicorn.main'"

**Expected behavior**
All tests should pass, 
A clear and concise description of what you expected to happen.

**Actual behavior**
From 0.3.26, `uvicorn.main` changed how logging is performed and does not expose `get_logger()` anymore. The logging logic was moved to the `main()` command and out of `run()`.

**To Reproduce**
Install with Uvicorn >= 0.3.26 and run the tests.

**Screenshots/Traceback**
```python
$ pytest --cov=./
ImportError while loading conftest '/home/travis/build/bocadilloproject/bocadillo/tests/conftest.py'.
tests/conftest.py:7: in <module>
    from bocadillo import API
bocadillo/__init__.py:1: in <module>
    from .api import API
bocadillo/api.py:12: in <module>
    from uvicorn.main import get_logger, run
E   ImportError: cannot import name 'get_logger' from 'uvicorn.main' (/home/travis/virtualenv/python3.7.1/lib/python3.7/site-packages/uvicorn/main.py)
The command "pytest --cov=./" exited with 4.
```

**Specs**
- OS: Ubuntu (TravisCI)
- Python Version: 3.6, 3.7
- Bocadillo Version: 0.10.0

**Additional context**
- [Build #744](https://travis-ci.org/bocadilloproject/bocadillo/builds/482593793)
- Blocks #103

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -14,6 +14,15 @@ As a result, we strongly recommend you read this document carefully before upgra
 
 ## [Unreleased]
 
+### Changed
+
+- Now requires `uvicorn>=0.3.26`.
+
+### Fixed
+
+- Fixed a bug that caused an `ImportError` when importing from
+`bocadillo.api` using `uvicorn<0.3.26`.
+
 ## [v0.10.0] - 2019-01-17
 
 ### Added

--- a/Pipfile
+++ b/Pipfile
@@ -5,7 +5,7 @@ name = "pypi"
 
 [packages]
 starlette = "*"
-uvicorn = "*"
+uvicorn = ">=0.3.26"
 parse = "*"
 "jinja2" = "*"
 whitenoise = "*"

--- a/bocadillo/api.py
+++ b/bocadillo/api.py
@@ -1,6 +1,6 @@
 import os
 from functools import partial
-from typing import Any, Dict, List, Optional, Type, Union, Callable, Tuple
+from typing import Any, Callable, Dict, List, Optional, Tuple, Type, Union
 
 from starlette.middleware.cors import CORSMiddleware
 from starlette.middleware.gzip import GZipMiddleware
@@ -9,22 +9,23 @@ from starlette.middleware.lifespan import LifespanMiddleware
 from starlette.middleware.trustedhost import TrustedHostMiddleware
 from starlette.middleware.wsgi import WSGIResponder
 from starlette.testclient import TestClient
-from uvicorn.main import get_logger, run
+from uvicorn.config import get_logger
+from uvicorn.main import run
 from uvicorn.reloaders.statreload import StatReload
 
 from .app_types import (
     ASGIApp,
     ASGIAppInstance,
-    Scope,
-    Receive,
-    Send,
-    EventHandler,
     ErrorHandler,
+    EventHandler,
+    Receive,
+    Scope,
+    Send,
 )
 from .compat import WSGIApp
 from .constants import DEFAULT_CORS_CONFIG
 from .error_handlers import error_to_text
-from .errors import ServerErrorMiddleware, HTTPErrorMiddleware, HTTPError
+from .errors import HTTPError, HTTPErrorMiddleware, ServerErrorMiddleware
 from .media import Media
 from .meta import DocsMeta
 from .recipes import RecipeBase
@@ -525,16 +526,14 @@ class API(TemplatesMixin, metaclass=DocsMeta):
         if debug:
             self.debug = True
             reloader = StatReload(get_logger(log_level))
-            reloader.run(
-                run,
-                {
-                    "app": self,
-                    "host": host,
-                    "port": port,
-                    "log_level": log_level,
-                    "debug": self.debug,
-                    **kwargs,
-                },
-            )
+            kwargs = {
+                "app": self,
+                "host": host,
+                "port": port,
+                "log_level": log_level,
+                "debug": self.debug,
+                **kwargs,
+            }
+            reloader.run(run, kwargs)
         else:
             _run(self, host=host, port=port, **kwargs)

--- a/setup.py
+++ b/setup.py
@@ -23,7 +23,7 @@ setuptools.setup(
     package_data={"bocadillo": ["assets/*"]},
     install_requires=[
         "starlette",
-        "uvicorn",
+        "uvicorn>=0.3.26",
         "jinja2",
         "whitenoise",
         "requests",

